### PR TITLE
chore: add dynamic shapes section in the resnet tutorial

### PR DESCRIPTION
**Pull request by peri044**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: examples/dynamo/vgg16_fp8_ptq.py [hunk 1]
```diff
@@ -216,16 +216,7 @@ def calibrate_loop(model):
...
-        # Compile the model with Torch-TensorRT Dynamo backend
-        input_tensor = images.cuda()
-        exp_program = torch.export.export(model, (input_tensor,))
-        trt_model = torchtrt.dynamo.compile(
-            exp_program,
-            inputs=[input_tensor],
-            enabled_precisions={torch.float8_e4m3fn},
-            min_block_size=1,
-            debug=False,
-        )
```
Comment (peri044) [inline examples/dynamo/vgg16_fp8_ptq.py:228]: Leave these changes as is (use torch.export workflow for the tutorial). I would say just change the title "Torch Compile VGG16 with FP8 and PTQ" -> "Deploying Torch-TRT programs in FP8 precision". In the comments, you can write the same workflow applies to torch.compile frontend as well.

File: examples/dynamo/vgg16_fp8_ptq.py
Comment (peri044) [inline examples/dynamo/vgg16_fp8_ptq.py:1]: Change the description to - This script demonstrates the Torch-TensorRT dynamo workflow on a VGG16 model with FP8 precision

File: docsrc/user_guide/dynamic_shapes.rst [hunk 2]
```diff
@@ -5,6 +5,10 @@ Dynamic shapes with Torch-TensorRT
 
 By default, you can run a pytorch model with varied input shapes and the output shapes are determined eagerly.
 However, Torch-TensorRT is an AOT compiler which requires some prior information about the input shapes to compile and optimize the model.
+
+Dynamic shapes with ir=dynamo
```
Comment (narendasan) [inline docsrc/user_guide/dynamic_shapes.rst:9]: rename to `Dynamic shapes using torch.export (AOT)` or something to that effect

File: docsrc/user_guide/dynamic_shapes.rst [hunk 3]
```diff
@@ -30,168 +34,46 @@ Under the hood
...
 
-In the conversion to TensorRT, we use the user provided dynamic shape inputs.
-We perform shape analysis using dummy inputs (across min, opt and max shapes) and store the
-intermediate output shapes which can be used in case the graph has a mix of Pytorch
-and TensorRT submodules.
+In the conversion to TensorRT, the graph already has the dynamic shape information in the node's metadata which will be used during engine building phase.
 
-Custom Constraints
-------------------
+Custom dynamic shape constraints
```
Comment (narendasan) [inline docsrc/user_guide/dynamic_shapes.rst:48]: captialization

File: docsrc/user_guide/dynamic_shapes.rst [hunk 4]
```diff
@@ -30,168 +34,46 @@ Under the hood
...
-    trt_mod = torch_tensorrt.dynamo.compile(graph_module, inputs=dynamic_inputs, **compile_spec)
-
-Limitations
------------
-
-If there are operations in the graph that use the dynamic dimension of the input, Pytorch
-introduces ``torch.ops.aten.sym_size.int`` ops in the graph. Currently, we cannot handle these operators and
-the compilation results in undefined behavior. We plan to add support for these operators and implement
-robust support for shape tensors in the next release. Here is an example of the limitation described above
+Torch-TensorRT automatically sets the constraints during ``torch.export`` tracing by constructing
```
Comment (narendasan) [inline docsrc/user_guide/dynamic_shapes.rst:52]: > Torch-TensorRT attempts to automatically set the constraints...

File: docsrc/user_guide/dynamic_shapes.rst [hunk 5]
```diff
@@ -30,168 +34,46 @@ Under the hood
...
-    model_outputs = model(input, input2)
+    inputs = torch.randn((1, 3, 224, 224), dtype=float32)
+    # User can provide dynamic shapes based on Torchdynamo feedback
+    exp_program = torch.export.export(model, (inputs,), dynamic_shapes=<user_provided_dynamic_shapes>)
+    trt_gm = torch_tensorrt.dynamo.compile(exp_program, [inputs])
+    # Run inference
+    trt_gm(inputs)
 
 
 Dynamic shapes with ir=torch_compile
```
Comment (narendasan) [inline docsrc/user_guide/dynamic_shapes.rst:71]: Dynamic shapes with `torch.compile` (JIT)

File: docsrc/user_guide/dynamic_shapes.rst [hunk 6]
```diff
@@ -200,10 +82,12 @@ TRT engine. This limitation will be addressed in future versions of Torch-Tensor
 
     model = MyModel().eval().cuda()
     inputs = torch.randn((1, 3, 224, 224), dtype=float32)
-    trt_gm = torch_tensorrt.compile(model, ir="torch_compile", inputs)
+    # This indicates the dimension 0 is dynamic and the range is [1, 8]
+    torch._dynamo.mark_dynamic(inputs, 0, min=1, max=8)
+    trt_gm = torch_tensorrt.compile(model, ir="torch_compile", [inputs])
```
Comment (narendasan) [inline docsrc/user_guide/dynamic_shapes.rst:87]: might as well call `torch.compile(backend="tensorrt")` since thats probably more common

File: examples/dynamo/vgg16_fp8_ptq.py [hunk 7]
```diff
@@ -1,10 +1,10 @@
 """
 .. _vgg16_fp8_ptq:
 
-Torch Compile VGG16 with FP8 and PTQ
+Deploying Torch-TRT programs in FP8 precision
```
Comment (narendasan) [inline examples/dynamo/vgg16_fp8_ptq.py:4]: Deploy Quantized Models using Torch-TensorRT

File: examples/dynamo/vgg16_fp8_ptq.py [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
 ======================================================
 
-This script is intended as a sample of the Torch-TensorRT workflow with `torch.compile` on a VGG16 model with FP8 and PTQ.
+This script demonstrates the Torch-TensorRT dynamo workflow on a VGG16 model with FP8 precision.
```
Comment (narendasan) [inline examples/dynamo/vgg16_fp8_ptq.py:7]: > Here we demonstrate how to deploy a model quantized to FP8 using the Dynamo frontend of Torch-TensorRT

File: docsrc/user_guide/dynamic_shapes.rst [hunk 9]
```diff
@@ -30,168 +34,46 @@ Under the hood
...
------------
-
-If there are operations in the graph that use the dynamic dimension of the input, Pytorch
-introduces ``torch.ops.aten.sym_size.int`` ops in the graph. Currently, we cannot handle these operators and
-the compilation results in undefined behavior. We plan to add support for these operators and implement
-robust support for shape tensors in the next release. Here is an example of the limitation described above
+Torch-TensorRT automatically sets the constraints during ``torch.export`` tracing by constructing 
+`torch.export.Dim` objects with the provided dynamic dimensions accordingly. Sometimes, we might need to set additional constraints and Torchdynamo errors out if we don't specify them.
+If you have to set any custom constraints to your model (by using `torch.export.Dim`), we recommend exporting your program first before compiling with Torch-TensorRT.
+Please refer to this `documentation <https://pytorch.org/tutorials/intermediate/torch_export_tutorial.html#constraints-dynamic-shapes>`_ to export the Pytorch module with dynamic shapes.
```
[the remark this answers is not in the document]
Comment (peri044) [reply docsrc/user_guide/dynamic_shapes.rst:55]: added it now.

**Review by narendasan — APPROVED** [review]
LGTM
